### PR TITLE
Remove DO_KILL_NODE _FAILURE state

### LINKED ISSUE #5709: add regression test for unhandled job queue state
i missed out on adding a regression test for the unhandled job state that was described in #4809 and solved by #5667.

@berland found this [example test](https://github.com/equinor/ert/blob/main/tests/unit_tests/c_wrappers/res/job_queue/test_job_queue.py#L107) that one might be able to piggy-back on and adjust to produce the desired state that would trigger the changes covered by the PR.

it would at least be nice to have a test covering this.

### PR DESCRIPTION
**Pull request by xjules**
**Issue**
Resolves #5709 

**Approach**
Todo:
- [x] remove DO_KILL_NODE _FAILURE and replace it with DO_KILL
- [ ] make regression test

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/clib/lib/include/ert/job_queue/job_status.hpp [hunk 1]
```diff
@@ -90,7 +90,6 @@ typedef enum {
     JOB_QUEUE_RUNNING_EXIT_CALLBACK = 2048,
     JOB_QUEUE_STATUS_FAILURE = 4096,
     JOB_QUEUE_FAILED = 8192,
-    JOB_QUEUE_DO_KILL_NODE_FAILURE = 16384,
```
Comment (berland) [inline src/clib/lib/include/ert/job_queue/job_status.hpp:93]: Reading the inline docs above: `NB: the status count algorithm has a HARD assumption that these
  values are on the 2^N form - without holes in the series. ` this should fail HARD.

File: src/ert/job_queue/job_queue_node.py [hunk 2]
```diff
@@ -358,7 +358,7 @@ def _log_kill_thread_stopping_status(self) -> None:
     DONE_STATES = [
         JobStatus.SUCCESS,
         JobStatus.IS_KILLED,
-        JobStatus.DO_KILL_NODE_FAILURE,
+        JobStatus.DO_KILL,
```
Comment (berland) [inline src/ert/job_queue/job_queue_node.py:361]: I am puzzled that `DO_KILL` was not here already. Do you understand why?

File: src/clib/lib/include/ert/job_queue/job_status.hpp (hunk 1 above)
Comment (xjules) [reply src/clib/lib/include/ert/job_queue/job_status.hpp:93]: Will update the value then

File: src/ert/job_queue/job_queue_node.py (hunk 2 above)
Comment (xjules) [reply src/ert/job_queue/job_queue_node.py:361]: Me too, will investigate ...

**Comment by xjules** [comment]
NB: Currently put on hold since we might remove the entire checking the OK file:
https://github.com/equinor/ert/blob/1260f3519490028b703bee2dfe2449df1bca3428/src/clib/lib/job_queue/job_node.cpp#L361

**Comment by oyvindeide** [comment]
> This needs to be revisited.

So we should convert it to an issue? Or in this form?

**Comment by oyvindeide** [comment]
Due to refactor I think this is outdated, will close it. Feel free to reopen if that is not correct.

**Pull request closed without merging**